reswift-jobs/reswift-jobs/UI/JobWindowTitleBarController.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  JobWindowTitleBarController.swift
+//  reswift-jobs
+//
+//  Created by Jay Koutavas on 2/23/20.
+//  Copyright © 2020 Heynow Software. All rights reserved.
+//
+
+import Cocoa
+
+class JobWindowTitleBarController: NSTitlebarAccessoryViewController {
+
+    var store: JobStore?
+    
+    @IBAction func addEmployee(_ sender: AnyObject) {
+        let targetRow = store?.state.job.items.count ?? 0
+        store?.dispatch(InsertEmployeeAction(employee: Employee.empty, index: targetRow))
+        store?.dispatch(SelectionAction.select(row: targetRow))
```

Comment:
Here's the heart of this PR.

Reply:
One could easily argue that selection is not an action appropriate to an add button. It makes an assumption about UI rendering.

Reply:
aside from your argument above, in `store?.state.job.items.count ?? 0`, `nil` coalescing and defaulting to `0` is done merely because `store` _can_ be nil. If `store` is `nil`, the entire function is a no-op. I'd have written this:

```swift
@IBAction func addEmployee(_ sender: AnyObject) {
  guard let store = store else { return }
  let targetRow = store.state.job.items.count
  store.dispatch(InsertEmployeeAction(employee: Employee.empty, index: targetRow))
  store.dispatch(SelectionAction.select(row: targetRow))
}
```